instrumentation/opentelemetry-instrumentation-celery/tests/test_tasks.py

```diff
@@ -77,3 +78,14 @@ def test_task(self):
         self.assertNotEqual(consumer.parent, producer.context)
         self.assertEqual(consumer.parent.span_id, producer.context.span_id)
         self.assertEqual(consumer.context.trace_id, producer.context.trace_id)
+
+    def test_uninstrument(self):
+        CeleryInstrumentor().instrument()
+        CeleryInstrumentor().uninstrument()
+
+        result = task_add.delay(1, 2)
+        while not result.ready():
+            time.sleep(0.05)
```

Comment:
do we need a timeout here?

Reply:
add delay method has an async ([result](https://docs.celeryq.dev/en/stable/getting-started/first-steps-with-celery.html)), timeout was added to verify that no spans were created due to the uninstrument() rather than async issue